libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py

```diff
@@ -219,4 +224,8 @@ def load_data(
                 + text_content[span["offset"] + span["length"] :]
             )
 
+        # save the text content into markdown format
+        if self.cache_dir is not None:
+            with open(Path(self.cache_dir) / f"{file_name.stem}.md", "w") as f:
+                f.write(text_content)
```

Comment:
@jacky0218 I think you should perform the full Markdown save before this chunk

```
        for span in removed_spans:
            text_content = (
                text_content[: span["offset"]]
                + text_content[span["offset"] + span["length"] :]
            )
```

Since this chunk will remove the tables and figures to avoid duplicates in our database.